pkg/api/train.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright © 2021 Cisco and/or its affiliates. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/pkg/errors"
+
+	"github.com/banzaicloud/go-cruise-control/pkg/types"
+)
+
+const (
+	EndpointTrain types.APIEndpoint = "TRAIN"
+)
+
+type TrainRequest struct {
+	types.GenericRequest
+
+	// Timestamp in millisecond of the latest metrics sample used to train Cruise Control, current time will be used
+	// if this parameter is not specified.
+	End int64 `param:"end,omitempty"`
+	// Timestamp in millisecond of the earliest metrics sample used to train Cruise Control.
+	Start int64 `param:"start"`
+}
+
+func (s TrainRequest) Validate() error {
+	if s.Start < 1 {
+		return errors.New("timestamp for training start must be bigger then 0")
+	}
+
+	if s.End < 1 {
```

Comment:
I think this is wrong. According to the description, `0` is a valid value (the current time will be used instead of zero).
Though `omitempty` is there, the Go struct int value will be initialized to zero, therefore the validation fails. 
Or am I missing something here?

Reply:
Nope. You are right. Will fix this by adding the right defaults and using constraints `Start < -1` and `End < 0` for validation.

Reply:
Fixed.